### ISSUE TITLE
💅 Vue genetic type not detected in generic components 

### ISSUE THREAD

**Issue by dpash** [biomejs/biome#3360]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm"
  JS_RUNTIME_VERSION:           "v20.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  Recommended:                  false
  All:                          false
  Enabled rules:
  complexity/noBannedTypes
  suspicious/noCatchAssign
  suspicious/noPrototypeBuiltins
  suspicious/noUnsafeNegation
  style/useAsConstAssertion
  suspicious/noDebugger
  complexity/noMultipleSpacesInRegularExpressionLiterals
  suspicious/noMisleadingInstantiator
  suspicious/noGlobalAssign
  suspicious/useValidTypeof
  suspicious/noAssignInExpressions
  correctness/noConstantCondition
  suspicious/noDuplicateParameters
  suspicious/useGetterReturn
  correctness/noEmptyPattern
  suspicious/noExtraNonNullAssertion
  correctness/noUnusedLabels
  correctness/noUnreachableSuper
  suspicious/noExplicitAny
  suspicious/noCompareNegZero
  correctness/noSwitchDeclarations
  correctness/useIsNan
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  suspicious/noAsyncPromiseExecutor
  suspicious/noControlCharactersInRegex
  complexity/noUselessTypeConstraint
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  suspicious/noMisleadingCharacterClass
  correctness/noPrecisionLoss
  suspicious/noRedeclare
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noFallthroughSwitchClause
  suspicious/noDuplicateObjectKeys
  suspicious/noUnsafeDeclarationMerging
  complexity/noUselessThisAlias
  correctness/noUnreachable
  correctness/noInnerDeclarations
  suspicious/noDuplicateCase
  correctness/noSelfAssign
  suspicious/noEmptyBlockStatements
  suspicious/noShadowRestrictedNames
  correctness/useValidForDirection
  complexity/noUselessCatch
  correctness/noUnsafeFinally
  correctness/noNonoctalDecimalEscape
  correctness/noUndeclaredVariables
  suspicious/noImportAssign
  correctness/useYield
  style/noNamespace
  correctness/noGlobalObjectCalls
  correctness/noNewSymbol
  correctness/noEmptyCharacterClassInRegex
  suspicious/noClassAssign
  suspicious/noFunctionAssign

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUndeclaredVariables

### Playground link

https://codesandbox.io/p/sandbox/nostalgic-shape-qytz8v

### Expected result

No error on line 7 of `example.vue`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by dpash** [comment]
This is related to #2456 but that was only to not throw an error on generic components rather than parsing the value inside the `generic` attribute.

    <script setup lang="ts" generic="T = string">
    import { ref } from "vue";
    
    const example = ref<T | null>(null);

Currently, this gives a `The T variable is undeclared.` error

**Comment by Sec-ant** [comment]
At the moment we can only detect and parse the JavaScript code ***inside*** these `<script>` tags, the lint rule doesn't understand the `generic` attribute. I'm afraid to say this is expected and I see this as an enhancement/feature request because our Vue support at the moment is partial and very ad-hoc.

We're still dicussing how we'll support HTML-ish languags fully. We welcome you to add your inputs in #1726, so we can settle on an infrastructure that can take most of the usecases into account.

In the meantime, I think you can try disabling the rule or using this option (tbh I'm not sure whether it will work) https://biomejs.dev/reference/configuration/#javascriptglobals as a workaround.
